File: src/#100/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}
var isSameTree = function (p, q) {
  if (p === null && q === null) {
    return true
  }
  if (p !== null && q !== null) {
    return q.val === p.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right)
  }
  return false
}

File: src/#100/solution.go
package solution

type TreeNode struct {
	Val         int
	Right, Left *TreeNode
}

func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if p != nil && q != nil {
		return p.Val == q.Val && isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
	}
	return false
}
